http_proxy.go

```diff
@@ -628,22 +649,11 @@ func (hp *HTTPProxy) Addr() (addrs []string, ok bool) {
 }
 
 func (hp *HTTPProxy) Close() error {
-	// Close listeners first to prevent new connections.
 	var err error
 	for _, l := range hp.listeners {
 		if e := l.Close(); e != nil {
 			err = multierr.Append(err, e)
 		}
 	}
-
-	// Shutdown the proxy to stop serving requests.
-	if e := hp.proxy.Shutdown(context.Background()); e != nil {
-		err = multierr.Append(err, e)
-	}
-
-	if tr, ok := hp.transport.(*http.Transport); ok {
-		tr.CloseIdleConnections()
-	}
-
```

Comment:
So now there is a `Close` that closes listeners, but user still needs to cancel ctx. 
Let's hide `Close` method and use in it Run. This way all workflow will be in `Run`.

Reply:
After implementing it I think the Close method gives users possibility for cleanup without relying on GC. It's actually kind of sad that the variable is not freed when it goes out of scope i.e. test function ends but that's how it is.

I'm reverting the last commit.

Reply:
ok